tests/test_run.py

```diff
@@ -441,6 +441,83 @@ def test_run_weekly(self):
         assert_allclose(otpt.flow, [10.0], atol=1e-7)
         assert_allclose(vs.volume, [30.0], atol=1e-7)
 
+    def test_initial_volume(self):
+        """Test RollingVirtualStorage node behaviour when initial volume is less than max volume"""
+
+        model = pywr.core.Model(timestep="1D")
+
+        inpt = Input(model, "Input", max_flow=0.0)
+        lnk = Link(model, "Link")
+        inpt.connect(lnk)
+        otpt = Output(model, "Output", max_flow=10, cost=-10.0)
+        lnk.connect(otpt)
+
+        vs = pywr.core.RollingVirtualStorage(
+            model, "Licence", [lnk], timesteps=3, initial_volume=70.0, max_volume=100.0
+        )
+
+        model.setup()
+
+        assert_allclose(vs.volume, [70.0], atol=1e-7)
+
+        model.step()
+
+        assert_allclose(otpt.flow, [0.0], atol=1e-7)
+        assert_allclose(vs.volume, [85.0], atol=1e-7)
+
+        model.step()
+
+        assert_allclose(otpt.flow, [0.0], atol=1e-7)
+        assert_allclose(vs.volume, [100.0], atol=1e-7)
+
+        model.step()
+
+        assert_allclose(otpt.flow, [0.0], atol=1e-7)
+        assert_allclose(vs.volume, [100.0], atol=1e-7)
+
+    def test_initial_pc(self):
+        """Test RollingVirtualStorage node behaviour when initial volume percentage is less than 1.0"""
+
+        model = pywr.core.Model(timestep="1D")
+
+        inpt = Input(model, "Input", max_flow=0.0)
+        lnk = Link(model, "Link")
+        inpt.connect(lnk)
+        otpt = Output(model, "Output", max_flow=10, cost=-10.0)
+        lnk.connect(otpt)
+
+        from pywr.parameters import ConstantParameter
+
+        mx_vol = ConstantParameter(model, 100)
+        vs = pywr.core.RollingVirtualStorage(
+            model,
+            "Licence",
+            [lnk],
+            timesteps=3,
+            initial_volume=70.0,
```

Comment:
Can we test the case with zero initial volume here? Also what happens if initial volume > max volume?

Reply:
Added test for zero initial volume. When initial volume > max volume the `after` method on storage forces the volume back in bounds. However that extra volume is available for the first timestep. This test passes:

```python
model = pywr.core.Model(timestep="1D")

inpt = Input(model, "Input", max_flow=200.0)
lnk = Link(model, "Link")
inpt.connect(lnk)
otpt = Output(model, "Output", max_flow=200.0, cost=-10.0)
lnk.connect(otpt)

vs = pywr.core.RollingVirtualStorage(
    model,
    "Licence",
    [lnk],
    timesteps=3,
    initial_volume=200.0,
    max_volume=100.0,
)

model.setup()

assert_allclose(vs.volume, [200.0], atol=1e-7)

model.step()

assert_allclose(otpt.flow, [200.0], atol=1e-7)
assert_allclose(vs.volume, [0.0], atol=1e-7)
```
Where do you think this error should captured given that we can't rely on knowing the max volume during setup? maybe in the storage `before` method or in the solver `_solve_scenario` method when it checks for available volume..

Reply:
Is this just an issue for RollingVirtualStorage or does some weird happen with other Storage nodes if initial_volume > max_volume? I think if it is a general issue then we can merge this as is and open an issue to resolve the wider problem.

Reply:
Looks like weird stuff happens with the other storage nodes. This passes:

```python

model = Model(timestep="1D")

storage = Storage(model, "storage", inputs=1, outputs=0, max_volume=20.0, initial_volume=30.0)
otpt = Output(model, "output", max_flow=0, cost=-1)
storage.connect(otpt)

model.setup()

assert_allclose(storage.volume, [30.0], atol=1e-7)

model.step()

assert_allclose(otpt.flow, [0.0], atol=1e-7)
assert_allclose(storage.volume, [30.0], atol=1e-7)

model.step()

assert_allclose(otpt.flow, [0.0], atol=1e-7)
assert_allclose(storage.volume, [30.0], atol=1e-7)
```
Unlike the virtual storages, the additional capacity of storage nodes is not removed after first timestep as the adjustment block in the `after` method does not run.

I think this line in the solver avoids any weird stuff happening in the LP in this situation as the bound just gets set to 0.0:

```python
 ub = max(max_volume - storage._volume[scenario_index.global_id], 0.0) / timestep.days
```
If it could go negative, I assume it would try and force water out of the storage and fail to find a solution if it could not.

I agree that it would be better to deal with this as a separate issue.

Reply:
Ok. Are you happy with the changes I've made?